health/core.go

```diff
@@ -136,6 +147,7 @@ func (c *Core) HandleMessage(msg event.StreamMessage) {
 
 func (c *Core) handleSingleEvent(evt data.Event) {
 	streamID, ts := evt.StreamID(), evt.Timestamp()
+	c.lastEventTs = ts
```

Comment:
The whole stream health analyzer logic already relies heavily on these timestamps, as we use them to build the stream metrics, reorder events etc. It's expected and OK for the clocks on different machines to be skewed by a couple of seconds, and it'd be ok if that happens on this logic too. The threshold we use here is much higher though, of 10 minutes, so I don't think there would be a problem because of clock skew.

There is one problem that could happen on startup tho, which is in case the analyzer starts up and is not able to sync all of the past (24h) events during the startup period. Since we're using the events own timestamps now, this means that the analyzer will only be considered healthy and start receiving traffic once it got at least 1 event from the last 10 minutes. Maybe this will be a problem when bootstrapping new clusters? Gotta keep in mind to maybe turn off this feature through the config flag when setting up a new cluster (or just leave the analyzer crash looping, no biiig harm on that either). 

To avoid that problem from happening on regular rollouts though I'm increasing the startup period of analyzers to 10 minutes as well: https://github.com/livepeer/livepeer-infra/pull/858